src/routes.js

```diff
@@ -1417,6 +1420,101 @@ module.exports = async (
     sentPayment.write(paymentRequest);
   });
 
+  app.post("/api/lnd/trackpayment", async (req, res) => {
+    const { router } = LightningServices.services;
+    const { paymentHash, inflightUpdates = true } = req.body;
+    if (req.limituser) {
+      const health = await checkHealth();
+      if (health.LNDStatus.success) {
+        res.sendStatus(403);
+      } else {
+        res.status(500);
+        res.json({ errorMessage: "LND is down" });
+      }
+    }
+
+    logger.info("Tracking payment payment", { paymentHash, inflightUpdates });
+    const trackedPayment = router.trackPaymentV2({
+      payment_hash: paymentHash,
+      no_inflight_updates: !inflightUpdates
+    });
+
+    trackedPayment.on("data", response => {
+      if (response.payment_error) {
+        logger.error("TrackPayment Info:", response)
+        return res.status(500).json({
+          errorMessage: response.payment_error
+        });
+      }
+      
+      logger.info("TrackPayment Data:", response);
+      return res.json(response);
+    });
+
+    trackedPayment.on("status", status => {
+      logger.info("TrackPayment Status:", status);
+    });
+
+    trackedPayment.on("error", async err => {
+      logger.error("TrackPayment Error:", err);
+      const health = await checkHealth();
+      if (health.LNDStatus.success) {
+        res.status(500).json({
+          errorMessage: sanitizeLNDError(err.message)
+        });
+      } else {
+        res.status(500);
+        res.json({ errorMessage: "LND is down" });
+      }
+    });
+  });
+
+  app.post("/api/lnd/sendtoroute", async (req, res) => {
+    const { router } = LightningServices.services;
+    const { paymentHash, route } = req.body;
+    if (req.limituser) {
+      const health = await checkHealth();
+      if (health.LNDStatus.success) {
+        res.sendStatus(403);
+      } else {
+        res.status(500);
+        res.json({ errorMessage: "LND is down" });
+      }
+    }
+
+    router.sendToRoute({ payment_hash: paymentHash, route }, (err, data) => {
+      if (err) {
+        logger.error("SendToRoute Error:", err);
+        return res.status(400).json(err);
+      }
+
+      return res.status(200).json(data);
+    });
+  });
+
+  app.post("/api/lnd/estimateroutefee", async (req, res) => {
+    const { router } = LightningServices.services;
+    const { dest, amount } = req.body;
+    if (req.limituser) {
+      const health = await checkHealth();
+      if (health.LNDStatus.success) {
+        res.sendStatus(403);
+      } else {
+        res.status(500);
+        res.json({ errorMessage: "LND is down" });
+      }
+    }
+
+    router.estimateRouteFee({ dest, amt_sat: amount }, (err, data) => {
+      if (err) {
+        logger.error("EstimateRouteFee Error:", err);
+        return res.status(400).json(err);
```

Comment:
Looks more like a 500 code error

Reply:
t would be a 500 code error if the route is erroring out because of a bug that's occurring from the server's code, but it's sending out a 400 code error because the user's input is most likely what caused this error to occur

Reply:
most likely, so not 100% of times? Then it's a 500 because we can't know for sure.